yearn/test/system/curve-voter-proxy-strategy.test.js

```diff
@@ -0,0 +1,343 @@
+  // pool gauge.
+  const synthetixCurveRewardsAddress =
+    "0xAF379f0228ad0d46bB7B4f38f9dc9bCC1ad0360c"
+  // Address of the Synthetix Curve rewards contract owner.
+  const synthetixCurveRewardsOwnerAddress =
+    "0xb3726e69da808a689f2607939a2d9e958724fc2a"
+  // Address of the governance managing the StrategyProxy contract.
+  const strategyProxyGovernanceAddress =
+    "0xFEB4acf3df3cDEA7399794D0869ef76A6EfAff52"
+
+  // Name of the vault for tBTCv2 Curve pool.
+  const vaultName = "Curve tBTCv2 Pool yVault"
+  // Symbol of the vault for tBTCv2 Curve pool.
+  const vaultSymbol = "yvCurve-tBTCv2"
+  // Total deposit limit of the vault for tBTCv2 Curve pool.
+  const vaultDepositLimit = to1ePrecision(300, 15)
+  // Amount of the deposit made by the depositor.
+  const vaultDepositAmount = to1ePrecision(300, 15)
+
+  let vaultGovernance
+  let tbtcCurvePoolLPToken
+  let vaultDepositor
+  let tbtcCurvePoolGaugeReward
+  let tbtcCurvePoolGaugeRewardDistributor
+  let synthetixCurveRewards
+  let synthetixCurveRewardsOwner
+  let strategyProxyGovernance
+  let vault
+  let strategy
+
+  before(async () => {
+    await resetFork(12786839)
+
+    // Setup roles.
+    vaultGovernance = await ethers.getSigner(0)
+    vaultDepositor = await impersonateAccount(
+      tbtcCurvePoolLPTokenHolderAddress,
+      vaultGovernance
+    )
+    strategyProxyGovernance = await impersonateAccount(
+      strategyProxyGovernanceAddress
+    )
+    tbtcCurvePoolGaugeRewardDistributor = await impersonateAccount(
+      tbtcCurvePoolGaugeRewardDistributorAddress,
+      vaultGovernance
+    )
+    synthetixCurveRewardsOwner = await impersonateAccount(
+      synthetixCurveRewardsOwnerAddress,
+      vaultGovernance
+    )
+
+    // Get tBTC v2 Curve pool LP token handle.
+    tbtcCurvePoolLPToken = await ethers.getContractAt(
+      "IERC20",
+      tbtcCurvePoolLPTokenAddress
+    )
+
+    // Setup Synthetix staking rewards to provide Curve pool's gauge additional
+    // rewards.
+    await setupSynthetixRewards()
+
+    // Get a handle to the Yearn registry.
+    const registry = await ethers.getContractAt(
+      "IYearnRegistry",
+      registryAddress
+    )
+
+    // Always use the same vault release to avoid test failures in the future.
+    // Target release index is taken from the deployed Yearn registry contract.
+    // We aim for version 0.4.2 (see BaseStrategy.apiVersion()) whose index is 9.
+    const targetReleaseIndex = 9
+    const releaseDelta = (await registry.numReleases()) - 1 - targetReleaseIndex
+
+    // Deploy a new experimental vault accepting tBTC v2 Curve pool LP tokens.
+    const tx = await registry.newExperimentalVault(
+      tbtcCurvePoolLPToken.address,
+      vaultGovernance.address,
+      vaultGovernance.address, // set governance to be the guardian as well
+      vaultGovernance.address, // set governance to be the rewards target as well
+      vaultName,
+      vaultSymbol,
+      releaseDelta
```

Comment:
Not sure if I'm following this `releaseDelta`. There's another `newExperimentalVault()` which won't take this param and also all the tests pass. From the equation on line 117 the result is `0`, but if you change it changes to other number because of `registry.numReleases()` it will fail.

Reply:
Yeah, that needs an additional explanation. The `newExperimentalVault` is defined in the Yearn's [`Registry`](https://github.com/yearn/yearn-vaults/blob/b60d2ef87645ef35a3d78a33d12591d70c7e9880/contracts/Registry.vy#L245) contract. This is a Vyper contract and it allows to set default function parameters. The `releaseDelta` param is the last parameter with a default value of `0`. However, once the Vyper code gets compiled, it produces two different signatures of the `newExperimentalVault` method - with and without the default parameter. Here we use the one which allows passing our own `releaseDelta`. We are doing this because we always need to deploy the same version of the vault contract. If new releases are added by the Yearn team, we are safe because we'll provide a right delta pointing to the same release we're using from the beginning.